### PR TITLE
Fix instruction inheritance issues

### LINKED ISSUE #1892: Allow inheritance of QuantumRegister and ClassicalRegister
Attempting to inherit from QuantumRegister or ClassicalRegister currently causes type errors in `instruction.py`'s constructor. There may be other areas that require updates to make this functionality viable.

### PR DESCRIPTION
**Pull request by Woody1193**
Fixes #1892

### Summary
Modified the way in which type-checking is done in instruction.py to allow QuantumRegister and ClassicalRegister to be inherited from.

### Details and comments
This PR is a repeat as I mistakenly attempted to merge with stable.

Type-equality checking is brittle and should be avoided except in the case where inheritance is to be disregarded for a class. That appears to be the case for neither QuantumRegister nor ClassicalRegister.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,10 +16,20 @@ The format is based on `Keep a Changelog`_.
   - **Fixed**: for any bug fixes.
   - **Security**: in case of vulnerabilities.
 
-
 `UNRELEASED`_
 =============
 
+
+`0.7.2`_ - 2019-03-05
+=====================
```
Comment (mtreinish) [inline CHANGELOG.rst]: This needs to be removed this isn't a 0.7.2 release and we also didn't push a 0.7.2 release on Tuesday.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -16,10 +16,20 @@ The format is based on `Keep a Changelog`_.
...
+
+
+Fixed
+-----
+
+- Fixed a bug whereby inheriting from QuantumRegister or ClassicalRegister
+  would cause a QiskitError in instruction.py
```
Comment (mtreinish) [inline CHANGELOG.rst]: This should be moved to the existing unreleased Fixed section which is below and already has a few items in the list. Also you can add `(#1908)` or ` (#1892)` to the end of the line.

File: CHANGELOG.rst (hunk 1 above)
Comment (Woody1193) [reply CHANGELOG.rst]: Thanks for the feedback. I'll get that updated

File: CHANGELOG.rst (hunk 2 above)
Comment (Woody1193) [reply CHANGELOG.rst]: Can do

File: CHANGELOG.rst (hunk 1 above)
Comment (Woody1193) [reply CHANGELOG.rst]: @mtreinish I have updated the changelog to reflect your feedback.

**Pull request merged**